src/newtondynamics_physics/ospnewton.h

```diff
@@ -0,0 +1,124 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2021 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include <osp/Active/SysPhysics.h>
+#include <osp/Active/ActiveScene.h>
+
+#include <Newton.h>
+
+namespace ospnewton
+{
+
+/**
+ * @brief Associates an entity with a NewtonBody and manages its lifetime
+ */
+struct ACompNwtBody
+{
+
+    ACompNwtBody(NewtonBody const* pBody) noexcept
```

Comment:
for such a simple class, is it necessary to use a private member variable?

you could use aggregate initialization, and completely skip the need for a constructor, and get auto-generated move constructors at the same time.

Reply:
Hmm, I guess so. Raw pointers were used before, and accidentally changing it is no longer an issue with unique_ptr. How about just making this entire struct a `using ACompNwtBody_t = std::unique_ptr<NewtonBody const, Deleter> m_pBody;`?

Reply:
Id' be perfectly happy with that.